docs/proposals/secret-manager-store-api.md

```diff
@@ -0,0 +1,62 @@
+# Secrets Manager Store Proposal
+
+**Author**: Emmanuel Bakare (@emmanuel-kubermatic)
+
+**Status**: Draft proposal; prototype in progress.
+
+## Goals
+The proposed fix for the issue [6671](https://github.com/kubermatic/kubermatic/issues/6671) 
+
+- Adds support for vault related read access within the kubermatic source
+- Polling support for SSH and Secrets within the Kubermatic source
+- Revocation and TTL-based access for SSH
+
+## Non-Goals
+Adds support for vault access and a central store for accessing data
+
+## Motivation and Background
+
+Currently, there's a user ssh agent which manages credentials via fsnotify events and kubernetes secrets.
+
+This approach is great but adds a limitation where the end user has to manually copy credentials to ssh secrets or add them via the API.
+
+The new approach documented within this proposal adds the ability to hard-code authorized keys but rather, add another option to lease out ssh certificates via vault which can be used to access the necessary clusters.
```

Comment:
Actually, the solution is not strictly coupled with Vault. We could just mention that we will add support for [TrustedUserCAKeys](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/deployment_guide/sec-distributing_and_trusting_ssh_ca_public_keys), it's not something Vault specific, even if Vault provides a convenient way of storing CA cert and signing keys:
https://www.vaultproject.io/docs/secrets/ssh/signed-ssh-certificates

Reply:
I've updated that in the proposal